src/main/java/MealDemo/service/MealPlanningServiceImpl.java

```diff
@@ -57,30 +63,24 @@ public void saveNewMeal(Meal meal) {
         System.out.println("meal saved! ");
     }
 
-
-    // DELETE
     @Override
     public boolean deleteMeal(int mealId) {
-        boolean isDeleted;
-        // find
-        Optional<Meal> tempMeal = findMealById(mealId);
+
+       Optional<Meal> tempMeal = mealRepository.findById(mealId);
         if (tempMeal.isEmpty()){
             System.out.println("Could not find meal");
-            isDeleted = false;
         } else {
             mealRepository.deleteById(mealId);
             System.out.println("Meal Deleted");
-            isDeleted = true;
         }
 
-        return isDeleted;
+        return tempMeal.isPresent();
     }
 
-    // UPDATE
     @Override
     public void updateMeal(int mealId, Meal meal) {
-        // find meal
-        Meal existingMeal = mealRepository.getById(mealId);
```

Comment:
use the repository here. I.e. undo this change. The reason is that often service methods will often grow to have side effects. Repo/Dao methods are clean. They do ONE thing and ONE thing only. Get some data, change some data, create some data . . . you interested in only doing the one thing so use the Repo method. Only use the service method if you want to risk calling future side effect . . . which ideally you shouldn't.